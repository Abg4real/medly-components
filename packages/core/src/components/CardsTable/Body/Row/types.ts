import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    onClick?: () => void;
    withWhiteBackground: boolean;
}
