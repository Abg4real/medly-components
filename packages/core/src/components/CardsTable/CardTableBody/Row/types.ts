import { Props as CardProps } from '../../../Card/types';

export interface Props extends CardProps {
    disabled?: boolean;
    withLightTheme: boolean;
}
