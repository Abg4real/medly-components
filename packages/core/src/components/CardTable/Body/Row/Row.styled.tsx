import { styled } from '@medly-components/utils';
import { Props } from './types';

export const Row = styled('tr').attrs(({ withWhiteBackground, theme: { cardTable } }: Props) => ({
    bgTheme: withWhiteBackground ? 'lightBackground' : 'darkBackground',
    cardTable
}))<Props & { bgTheme: 'lightBackground' | 'darkBackground' }>`
    display: grid;
    grid-template-columns: ${({ gridTemplateColumns }) => gridTemplateColumns};
    cursor: ${({ onClick, isRowClickDisabled }) => (onClick && !isRowClickDisabled ? 'pointer' : 'inherit')};
    box-sizing: border-box;
    border-radius: 0.8rem;
    margin-bottom: 0.8rem;
    background-color: white;
    transition: box-shadow 100ms ease-out;
    border: 0.1rem solid ${({ bgTheme, cardTable }) => cardTable[bgTheme].borderColor};
    box-shadow: ${({ bgTheme, cardTable }) => cardTable[bgTheme].boxShadow};

    &:hover {
        box-shadow: ${({ bgTheme, cardTable }) => cardTable[bgTheme].hoverBoxShadow};
    }

    :nth-child(odd) {
        background: ${({ bgTheme, cardTable }) => cardTable[bgTheme].oddRowBgColor};
    }
    :nth-child(even) {
        background: ${({ bgTheme, cardTable }) => cardTable[bgTheme].evenRowBgColor};
    }
`;


export const NoResult = styled(Row)`
    display: flex;
    padding: 1.0rem 0;
`;
