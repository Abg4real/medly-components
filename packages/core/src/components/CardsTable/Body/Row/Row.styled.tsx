import { styled } from '@medly-components/utils';
import Card from '../../../Card';
import { Props } from './types';

export const Row = styled(Card)<Props>`
    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'grab' : 'inherit')};
    box-sizing: border-box;
    border-radius: 0.8rem;
    border: ${({ withLightTheme }) => withLightTheme ? 'default' : 'none'};
    box-shadow: ${({ withLightTheme }) => withLightTheme ? 'none' : '0 0.2rem 0.8rem rgba(176,188,200,0.2)'};
        
    &:hover {
        box-shadow: ${({ withLightTheme }) => withLightTheme ? '0 0.2rem 0.4rem rgba(176,188,200,0.60)' : '0 0.4rem 0.8rem rgba(176,188,200,0.60)'};
    };

    & > * {
        background-color: ${({ theme, disabled }) => `${disabled ? theme.cardsTable.rowDisabledColor : 'inherit'}`};
    };

    :nth-child(2n) {
        background: ${({ theme, withLightTheme}) => withLightTheme ? theme.cardsTable.rowDisabledColor : theme.cardsTable.rowBackgroundColor};
    };
`;