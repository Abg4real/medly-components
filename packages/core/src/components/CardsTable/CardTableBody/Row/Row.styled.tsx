import { styled } from '@medly-components/utils';
import Card from '../../../Card';
import { Props } from './types';

export const Row = styled(Card)<Props>`
    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'grab' : 'inherit')};
    box-shadow: none;
    box-sizing: border-box;
    border-radius: 0.8rem;
    border: ${({ withLightTheme }) => withLightTheme ? 'default' : 'none'};

    &:hover {
        box-shadow: ${({ withLightTheme, theme }) => withLightTheme ? `0px 2px 4px ${theme.colors.grey[400]}` : `0px 4px 8px ${theme.colors.grey[400]}`};
    };

    & > * {
        background-color: ${({ theme, disabled }) => `${disabled ? theme.table.rowDisabledColor : 'inherit'}`};
    };

    :nth-child(2n) {
        background: ${({ theme, withLightTheme}) => withLightTheme ? `${theme.colors.grey[100]}` : theme.table.rowBackgroundColor};
    };
`;


