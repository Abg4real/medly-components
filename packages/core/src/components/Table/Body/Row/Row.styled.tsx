import { css, styled } from '@medly-components/utils';
import { GridTemplateProps } from '../../types';
import { Props } from './types';

const selectedBorder = css`
    &&& {
        &::before {
            content: '';
            left: 0;
            top: -0.05rem;
            z-index: 10;
            height: 100%;
            width: 0.4rem;
            position: absolute;
            background-color: ${({ theme }) => theme.table.row.selectedBorderColor};
        }
    }
`;

export const Row = styled('tr').attrs(({ gridTemplateColumns }: GridTemplateProps) => ({
    style: {
        gridTemplateColumns
    }
}))<Props>`
    display: grid;
    position: relative;
    align-items: center;
    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'pointer' : 'inherit')};

    &:nth-child(odd) {
        & > * {
            background-color: ${({ theme, isSelected, disabled }) =>
                theme.table.row.bgColor[disabled ? 'disabled' : isSelected ? 'selected' : 'odd']};
        }
    }

    &:nth-child(even) {
        & > * {
            background-color: ${({ theme, isSelected, disabled }) =>
                theme.table.row.bgColor[disabled ? 'disabled' : isSelected ? 'selected' : 'even']};
        }
    }

    &:not(:last-child) {
        & > * {
            border-bottom: 1px solid ${({ theme }) => theme.table.borderColor};
        }
    }

    ${({ isSelected }) => isSelected && selectedBorder}
`;

export const NoResult = styled(Row)`
    text-align: center;
    line-height: 30px;
`;
