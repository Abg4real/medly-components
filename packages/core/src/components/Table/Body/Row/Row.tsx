import React, { useCallback, useContext, useState } from 'react';
import { GroupCell } from '../../GroupCell';
import { getGridTemplateColumns } from '../../helpers';
import { TablePropsContext } from '../../TableProps.context';
import { TableColumnConfig } from '../../types';
import Cell from '../Cell';
import { ExpandedCell } from '../Cell/Cell.styled';
import * as Styled from './Row.styled';
import { Props } from './types';

export const Row: React.FC<Props> = React.memo(props => {
    const [isExpanded, setExpansionState] = useState(false),
        { data, showShadowAfterFrozenElement, selectedRowIds, onRowSelection, addColumnMaxSize, ...restProps } = props,
        { columns, isLoading, onRowClick, rowIdentifier, rowClickDisableKey, rowSelectionDisableKey } = useContext(TablePropsContext);

    const handleExpansionIconClick = useCallback(() => setExpansionState(val => !val), []),
        stopPropagation = useCallback((e: React.MouseEvent) => e.stopPropagation(), []),
        getCells = (rowData: any = {}, configs: TableColumnConfig[] = columns, field = '') =>
            configs.map(config => {
                const fieldName = `${field && `${field}.`}${config.field}`;

                return config.children ? (
                    <GroupCell
                        as={field ? 'div' : 'td'}
                        key={config.field}
                        hidden={config.hidden}
                        gridTemplateColumns={getGridTemplateColumns(config.children)}
                    >
                        {getCells(rowData[config.field], config.children, config.field)}
                    </GroupCell>
                ) : (
                    <Cell
                        key={fieldName}
                        as={field ? 'div' : 'td'}
                        isLoading={isLoading}
                        isRowClickDisabled={rowData[rowClickDisableKey]}
                        isRowSelectionDisabled={rowData[rowSelectionDisableKey]}
                        data={rowData[config.field]}
                        rowId={rowData[rowIdentifier]}
                        dottedFieldName={fieldName}
                        isExpanded={isExpanded}
                        onExpansionIconClick={handleExpansionIconClick}
                        showShadowAtRight={showShadowAfterFrozenElement}
                        isRowSelected={selectedRowIds.includes(rowData[rowIdentifier])}
                        {...{ config, onRowSelection, addColumnMaxSize }}
                    />
                );
            }),
        handleRowClick = (rowData: any) =>
            !isLoading && onRowClick && !rowData[rowClickDisableKey] ? () => onRowClick(rowData) : undefined;

    return (
        <>
            <Styled.Row
                {...restProps}
                disabled={data[rowClickDisableKey]}
                onClick={handleRowClick(data)}
                isSelected={!isLoading && selectedRowIds.includes(data[rowIdentifier])}
                gridTemplateColumns={getGridTemplateColumns(columns)}
            >
                {getCells(data)}
                {isExpanded && <ExpandedCell onClick={stopPropagation}>Something</ExpandedCell>}
            </Styled.Row>
        </>
    );
});
Row.displayName = 'Row';
