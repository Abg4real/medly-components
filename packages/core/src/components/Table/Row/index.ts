import { NoResult, Row } from './Row.styled';

export default Row;

export { NoResult };
