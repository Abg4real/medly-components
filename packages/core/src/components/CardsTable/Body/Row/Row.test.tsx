import { render } from '@test-utils';
import React from 'react';
import { Row } from './Row.styled';

describe('Row', () => {
    it('should render row properly', () => {
        const { container } = render(
            <tbody>
                <Row disabled={true} withLightTheme={true}>
                    <td>Demo</td>
                </Row>
            </tbody>
        );
        expect(container).toMatchSnapshot();
    });
});
