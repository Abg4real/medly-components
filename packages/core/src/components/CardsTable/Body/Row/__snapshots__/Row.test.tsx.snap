// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render row properly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 5px;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  background-color: #ffffff;
  padding: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: 1px solid #dfe4e9;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

.c1 {
  cursor: not-allowed;
  box-shadow: none;
  box-sizing: border-box;
  border-radius: 0.8rem;
  border: default;
}

.c1:hover {
  box-shadow: 0 0.2rem 0.4rem rgba(176,188,200,0.60);
}

.c1 > * {
  background-color: #eff2f4;
}

.c1:nth-child(2n) {
  background: #eff2f4;
}

<div>
  <div
    class="c0 c1"
    disabled=""
  >
    Dummy
  </div>
</div>
`;
